### PR TITLE
MAINT: refactor sqlite array type handling

### PR DESCRIPTION
**Pull request by GavinHuttley**
[CHANGED] make sure BytesIO instances are closed when
    finished.

### CODE REVIEW

**Comment by GavinHuttley** [comment]
codacy issue relates to Markdown formatting that's used by scriv for producing changelog. It's safely ignored.

**Review by rmcar17 — CHANGES_REQUESTED** [review]
Other than that relatively small comment, looks good to me!

Should we type those functions as well as `array_to_sqlite` and `sqlite_to_array` while we're at it? Didn't think about it in my initial implementation.

**Comment by rmcar17** [comment]
The pull request reminds me I should detail the solution in the issue (#1776) linked to by the warning.

File: tests/test_core/test_annotation_db.py [hunk 1]
```diff
@@ -925,7 +925,9 @@ def test_write(gb_db, tmp_path):
 
 
 def convert_to_old_np_format(data):
-    return numpy.load(io.BytesIO(data)).tobytes()
+    with io.BytesIO(data) as data:
```
[the remark this answers is not in the document]
Comment (GavinHuttley) [reply tests/test_core/test_annotation_db.py]: I agree we need to add type hints. I have also taken your suggestion of renaming the variables in the test function.

**Review by rmcar17 — APPROVED** [review]
Looks good to me!

**Pull request merged**